src/core/load-dropin.c

```diff
@@ -19,51 +19,245 @@
 
 
 #include "conf-parser.h"
+#include "fs-util.h"
 #include "load-dropin.h"
 #include "load-fragment.h"
 #include "log.h"
+#include "path-util.h"
+#include "stat-util.h"
 #include "strv.h"
 #include "unit-name.h"
 #include "unit.h"
 
+
+typedef struct DependencyConsumerData {
+        Unit *unit;
+        int depth;
+} DependencyConsumerData;
+
+typedef struct DropinSymlink {
+        Unit *unit;
+        UnitDependency dependency;
+        int depth;              /* path depth used when merging 2 dropins */
+        bool is_mask;           /* does the symlink points to /dev/null ? */
+} DropinSymlink;
+
+static void dropin_hash_func(const void *p, struct siphash *state) {
+        const DropinSymlink *ds = p;
+
+        siphash24_compress(&ds->unit, sizeof(ds->unit), state);
```

Comment:
hmm, if i got this right, then the hash table is indexed by the combination of unit plus dependency. It would make a ton of sense hence, to also add the dependency to the hash value here, simply by adding shiphash24_compress() on it too...